jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/telemetry/CodeWhispererCodeCoverageTracker.kt

```diff
@@ -52,7 +53,7 @@ abstract class CodeWhispererCodeCoverageTracker(
 
     @Synchronized
     fun activateTrackerIfNotActive() {
-        if (!isTelemetryEnabled() || isActive.get()) return
+        if (!isTelemetryEnabled() || !CodeWhispererExplorerActionManager.getInstance().hasAcceptedTermsOfService() || isActive.get()) return
```

Comment:
In fact this is a problem not only for the calculation of this codePercentage metric but for all CodeWhisperer logics, we want the various listeners to not take effect when CWSPR is disabled. We can put this check in the documentChanged() event in `CodeWhispererEditorListener`. There needs to be other changes but for the scope of this issue this will be enough.

Reply:
And replace `isActive.get()` with `isTrackerActive()` here?

Reply:
> 

yea make sense

Reply:
yea make sense we also need to add this condition in all other cwspr logics. Yes, let me put this in editorListener